### PR TITLE
0.9.0 changelog

### PR DESCRIPTION
**Pull request by nbgl**
API overhaul changelog.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,30 +1,141 @@
+0.9.0
+=====
+
+This release contains a major overhaul of Anonlink’s API.
```
Comment (hardbyte) [inline CHANGELOG.rst]: Mention the headline feature *multiparty support*

File: CHANGELOG.rst [hunk 2]
```diff
@@ -1,30 +1,141 @@
...
+  - Multiple files with serialised candidate pairs can be merged without loading everything into memory at once.
+- Introduce type annotations in the new API.
+
+Minor changes
+-------------
+- Automatically test on Python 3.7.
+- Remove support for Python 3.5 and below.
+- Update Clkhash dependency to 0.11.
+- Minor documentation and style in ``anonlink.concurrency``.
+- Provide a convenience function for generating a valid candidate pairs from a chunk.
```
Comment (hardbyte) [inline CHANGELOG.rst]: generating ~a~ valid candidate pairs from a chunk.

File: CHANGELOG.rst (hunk 1 above)
Comment (nbgl) [reply CHANGELOG.rst]: Done!

File: CHANGELOG.rst (hunk 2 above)
Comment (nbgl) [reply CHANGELOG.rst]: I good grammer.

File: CHANGELOG.rst [hunk 3]
```diff
@@ -1,30 +1,141 @@
+0.9.0
+=====
+
+This release contains a major overhaul of Anonlink’s API.
+
+The changes are all additive, so the previous API continues to work. That API has now been deprecated and will be removed in a future release.
```
[the remark this answers is not in the document]
Comment (nbgl) [reply CHANGELOG.rst]: There isn't an issue I can link to, but I'll include a deprecation timeline.

**Comment by nbgl** [comment]
@hardbyte, could you please have a glance at the small changes I made since your review?

**Pull request merged**